### ISSUE TITLE
AttributeError: 'list' object has no attribute 'canvas'

### ISSUE THREAD

**Issue by abnitchauhan** [mpld3/mpld3#460]
I am trying to save the data in json or in html(one way or the other). But during both the processes I am getting the Following ERROR : 
```
     if fig.canvas is None:
AttributeError: 'list' object has no attribute 'canvas'
```
I have included the all the required libraries , but still not able to figure out , what the problem is .

Here is the Code Which  I am using.
```
plt.subplots(1, 1, figsize=(8, 2))
ecg = X
#fig=plt.figure()
alt = np.arange(len(ecg))/125
fig= plt.plot(alt,ecg)
mpld3.save_html(fig,"test.html")
mpld3.fig_to_html(fig,template_type="simple")
mpld3.fig_to_dict(fig)
mpld3.show()
```

I am getting the graph If I am accessing it dircetly without trying to save. But I want to save the figure so that I can use it in webpage, which I am not able to do so.
Please Help me out.

**Comment by bryevdv** [comment]
There has been no substantive public work on this project in more than five years. Here is an (incomplete) list of actively maintained Python visualization projects you should consider instead:

* [Altair](https://altair-viz.github.io)
* [Bokeh](https://bokeh.org)
* [Matplotlib](https://matplotlib.org)
* [Plotly](https://plot.ly)
* [Pyviz](https://pyviz.org)

**Comment by ckerr-IDM** [comment]
@abnitchauhan @bryevdv Hi both -- that's not quite true, mpld3 is maintained on the fork https://github.com/sciris/mpld3, and we will be merging back into this version "soon" (we've had some staffing issues on our team which has delayed the merge).

@abnitchauhan , the reason for your error is that `plt.plot()` returns a "list of 2D plotted data", not a figure object. You had it almost right with what you had commented out! The following code should work:
```
plt.subplots(1, 1, figsize=(8, 2))
ecg = X
fig=plt.figure()
alt = np.arange(len(ecg))/125
plt.plot(alt,ecg)
mpld3.save_html(fig,"test.html")
mpld3.fig_to_html(fig,template_type="simple")
mpld3.fig_to_dict(fig)
mpld3.show()
```

**Comment by abnitchauhan** [comment]
@ckerr-IDM . Thank you for the above Answer that you provided . now the file is saved in html file as well as I am able to same file in json format.
But now , what I want is that to show the data in my webpage. there is no details provided about that in the gmpld3 github account regarding that.

**Comment by cliffckerr** [comment]
@abnitchauhan Yes it's not completely straightforward using pure mpld3, which is why we developed [Sciris](https://github.com/sciris/scirisweb). The "Hello Graph" example does exactly what I think you're after: create a graph and display it in a webpage:
Python app: https://github.com/sciris/scirisweb/blob/develop/examples/hellograph/app.py
HTML: https://github.com/sciris/scirisweb/blob/develop/examples/hellograph/index.html

You can do it without using the Sciris package, but it's going to be a bit more involved.

**Comment by vladh** [comment]
@abnitchauhan I'll go ahead and close this issue. We've released the latest version of mpld3 on PyPI (0.5.0), but I don't think mpld3 makes it straightforward to do what you're trying to do. However, feel free to reply to this issue if I can help further!
